docs/datasets/data-anonymization/index.md

```diff
@@ -0,0 +1,221 @@
+
+grounding_dino:
+  box_threshold: 0.1 # Threshold for the bounding box (float)
+  text_threshold: 0.1 # Threshold for the text (float)
+  nms_threshold: 0.1 # Threshold for the non-maximum suppression (float)
+
+open_clip:
+  score_threshold: 0.7 # Validity threshold for the OpenCLIP model (float
+
+bbox_validation:
+  iou_threshold: 0.9 # Threshold for the intersection over union (float), if the intersection over union is greater than this threshold, the object will be selected as inside the validation prompt
+```
+
+- `config/yolo_train.yaml`
+
+```yaml
+dataset:
+  input_dataset_yaml: "path/to/input/data.yaml" # Path to the config file of the dataset, which is created in the previous step
+
+yolo:
+  epochs: 100 # Number of epochs for the YOLOv8 model (int)
+  model: "yolov8x.pt" # Select the base model for YOLOv8 ('yolov8x.pt' 'yolov8l.pt', 'yolov8m.pt', 'yolov8n.pt')
+```
+
+- `config/yolo_anonymize.yaml`
+
+```yaml
+rosbag:
+  input_bag_path: "path/to/input.bag" # Path to the input ROS 2 bag file with 'mcap' or 'sqlite3' extension
+  output_bag_path: "path/to/output/folder" # Path to the output ROS 2 bag folder
+  output_save_compressed_image: True # Save images as compressed images (True or False)
+  output_storage_id: "sqlite3" # Storage id for the output bag file (`sqlite3` or `mcap`)
+
+yolo:
+  model: "path/to/yolo/model" # Path to the trained YOLOv8 model file (`.pt` extension) (you can download the pre-trained model from releases)
+  config_path: "path/to/input/data.yaml" # Path to the config file of the dataset, which is created in the previous step
+  confidence: 0.15 # Confidence threshold for the YOLOv8 model (float)
+
+blur:
+  kernel_size: 31 # Kernel size for the Gaussian blur (int)
+  sigma_x: 11 # Sigma x for the Gaussian blur (int)
+```
+
+## Usage
+
+The tool provides two options to anonymize images in ROS 2 bag files.
+
+### Option 1: Anonymize with Unified Model
+
+You should provide a single rosbag and tool anonymize images in rosbag with a unified model. The model is a combination
+of GroundingDINO, OpenCLIP, YOLOv8 and SegmentAnything. If you don't want to use pre-trained YOLOv8 model, you can
+follow the instructions in the second option to train your own YOLOv8 model.
+
+You should set your configuration in config/anonymize_with_unified_model.yaml file.
+
+```bash
+python3 main.py config/anonymize_with_unified_model.yaml --anonymize_with_unified_model
+```
+
+### Option 2: Anonymize Using the YOLOv8 Model Trained on a Dataset Created with the Unified Model
+
+#### Step 1: Create a Dataset
+
+Create an initial dataset with the unified model. You can provide multiple ROS 2 bag files to create a dataset. After
+running the following command, the tool will create a dataset in YOLO format.
+
+You should set your configuration in config/yolo_create_dataset.yaml file.
+
+```bash
+python3 main.py config/yolo_create_dataset.yaml --yolo_create_dataset
```

Comment:
YOLO config file is missing. Could you share a config file in release?

Reply:
It was set resolved, but I still got the following error when I tried to run this command:

```
Traceback (most recent call last):
  File "/home/mitsudome-r/autoware_rosbag2_anonymizer/main.py", line 81, in <module>
    yolo_create_dataset(
  File "/home/mitsudome-r/autoware_rosbag2_anonymizer/autoware_rosbag2_anonymizer/tools/yolo_create_dataset.py", line 28, in yolo_create_dataset
    unified_language_model = UnifiedLanguageModel(config_data, json_data)
  File "/home/mitsudome-r/autoware_rosbag2_anonymizer/autoware_rosbag2_anonymizer/model/unified_language_model.py", line 47, in __init__
    self.yolo_model_name = config["yolo"]["model"]
KeyError: 'yolo'
```

It seams like it is expecting yolo parameters defined in yolo_create_dataset.yaml, but it is not defined in the default configuration file.

Reply:
I apologize, I missed that. I have added the missing variable and also included an `if` statement to check if the YOLO model exists. If users haven't downloaded the pre-trained model, it will still work.

related commit: https://github.com/autowarefoundation/autoware_rosbag2_anonymizer/commit/9ea65e1c8a9359626c95cba2f1df68102ebd645b